### PR TITLE
New features

### PR DESCRIPTION
**Pull request by stamsarger**
I fixed the infamous hasOwnProperty bug (although you had already fixed it but neglected to push your code here), however some other bugs still remain. ModBot somehow crashes when it sees a link. I didn't fix that, lel. 🖕
New commands:
- autoban
- nickautoban
I commented more on them in the commit page.

### CODE REVIEW

File: hackchat.js [hunk 1]
```diff
@@ -30,6 +31,8 @@ var HackChatSession = (function (_super) {
         this.ws.on("message", function (data) {
             try {
                 var json = JSON.parse(data);
+                if (["valueof", "hasownproperty", 'tostring'].indexOf(String(json.nick).toLowerCase()) != -1)
```
Comment (ToastyStoemp) [inline hackchat.js:34]: not needed

File: main.js [hunk 2]
```diff
@@ -21,7 +20,7 @@ var users = {};
 var links = [];
 var connections = {};
 var afk = [];
-
+var chaton = [];
```
Comment (ToastyStoemp) [inline main.js:23]: Could use a better naming convention

File: main.js [hunk 3]
```diff
@@ -93,6 +92,29 @@ chat.on("chat", function(session, nick, text, time, isAdmin, trip) {
     }
 });
 
+chat.on('onlineSet', function(session, users) {
+    for (var i = 0; i < users.length; i++) {
+        if (userStats[users[i]] && userStats[users[i]].autoBan === true) {
```
Comment (ToastyStoemp) [inline main.js:97]: would not work, userStats is a combination of nick + trip

File: main.js [hunk 4]
```diff
@@ -115,10 +137,32 @@ chat.on("ratelimit", function() {
 });
 
 chat.on("onlineRemove", function(session, nick) {
-  if (afk.indexOf(nick) != -1)
-    afk.splice(afk.indexOf(nick), 1);
+    if (afk.indexOf(nick) != -1)
+        afk.splice(afk.indexOf(nick), 1);
+    chaton.splice(chaton.indexOf(nick), 1);
```
Comment (ToastyStoemp) [inline main.js:142]: what does this even do ?

File: main.js [hunk 5]
```diff
@@ -527,6 +571,47 @@ parseCommand = function(session, nick, message, isMod) {
             case "leave":
                 leave(session);
                 return;
+            case "autoban":
```
Comment (ToastyStoemp) [inline main.js:574]: No trip based auto ban,
No time based auto ban,
No analysis for similar users

File: hackchat.js (hunk 1 above)
Comment (stamsarger) [reply hackchat.js:34]: ok then

**Review by ToastyStoemp — COMMENTED** [review]
You could do like super easy fix: 
nick += " ";

File: main.js (hunk 3 above)
Comment (stamsarger) [reply main.js:97]: but i tried it and it worked O.o

**Pull request closed without merging**